### PR TITLE
py-cliff: update to 4.7.0 and add missing dependency

### PR DESCRIPTION
**Pull request by someuser12**
#### Description

`py-cliff` at its present version (4.4.0) misses the Python library Autopage. This PR adds the missing dependency (via the submission of the port `py-autopage`) and updates `py-cliff`.

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 10.13.6 17G14042 x86_64
Xcode 9.1 9B55

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by reneeotten — CHANGES_REQUESTED** [review]
this should be separated in three different commits please

File: python/py-cliff/Portfile [hunk 1]
```diff
@@ -16,21 +16,25 @@ long_description    cliff is a framework for building command line \
...
 
-python.versions     38 39 310 311
+python.versions     38 39 310 311 312
 
 if {${subport} ne ${name}} {
     depends_build-append \
                     port:py${python.version}-pbr
 
+    if {${python.version} < 310} {
+        depends_build-append  port:py${python.version}-importlib-metadata
```
Comment (reneeotten) [inline python/py-cliff/Portfile:30]: this should be "depends_lib"

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks for making this change @someuser12 !

One dependency type seems wrong to me and your first commit message isn't very clear. It should be "py-autopage: new port, version 0.5.2". 
Same for the second commit, it should be: "py-cmd2: add py312 subport"

File: python/py-cliff/Portfile (hunk 1 above)
Comment (someuser12) [reply python/py-cliff/Portfile]: Thanks for checking and suggesting the corrections @reneeotten.

**Pull request merged**